CONTRIBUTING.md

```diff
@@ -0,0 +1,121 @@
+
+To enforce linting and reduce review effort, you can use [black](https://pypi.org/project/black/) for linting. You can install via:
+```
+pip3 install black
+```
+To enforce linting of the code:
+```
+black <path-to-repository>/connaisseur
+```
+
+You can also test your changes locally. We recommend the following approach for running pytest in a docker container:
+```
+docker run -it --rm -v <path-to-repository>/connaisseur:/data --entrypoint=ash python:alpine
+cd data
+pip3 install -r requirements.txt
+pip3 install .
+cd connaisseur
+pytest --cov=connaisseur --cov-report=xml tests/
+```
+This will help you see  whether your changes fail the tests before committing any changes.
+
+> :information_source: **INFO** We believe that testing should not only ensure functionality, but also aim to test for expected security issues like injections and appreciate if security tests are added with new functionalities.
+
+The above testing provides unit testing. Before any PR can be merged, an integration test should be carried out whereby:
+- Connaisseur is successfully installed in a test cluster
+- a non-signed image is deployed to the cluster and denied
+- a signed image is deployed to the cluster and passed
+- Connaisseur is successfully uninstalled
+
+We hope to automate these tests in the future and also provide a test script for local testing with Minikube.
+
+### Signed Commits and Pull Requests
+A general introduction into signing commits can for example be found in the [With Blue Ink blog](https://withblue.ink/2020/05/17/how-and-why-to-sign-git-commits.html). For details on setting everything up for GitHub, please follow the steps in the [Documentation](https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification).
+
+Once you have generated your local GPG key, added it to your GitHub account and informed Git about it, you are setup to create signed commits. We recommend to configure Git to sign commits by default via:
+```bash
+git config commit.gpgsign true
+```
+This avoids forgetting to use the `-S` flag when committing changes. In case it happens anyways, you can always rebase to sign earlier commits:
+```bash
+git rebase -i master
+```
+You can then mark all commits that need to be signed as `edit` and sign them without any other changes via:
+```bash
+git commit -S --amend --no-edit
+```
+Finally, you force push to overwrite the unsigned commits via `git push -f`.
+
+To ensure that only signed commits get merged into master, the signatures should be checked when merging to master via:
```

Comment:
If possible in Github, I would go for pre-receive git-hooks. Otherwise we might also propose client-side git-hook

Reply:
Or we could enable [protected branches](https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/about-required-commit-signing)

Reply:
> Or we could enable protected branches

done